AuthenticatorShared/UI/Vault/ItemList/ItemList/ItemListProcessor.swift

```diff
@@ -429,15 +429,52 @@ extension ItemListProcessor: AuthenticatorKeyCaptureDelegate {
         _ captureCoordinator: AnyCoordinator<AuthenticatorKeyCaptureRoute, AuthenticatorKeyCaptureEvent>,
         key: String
     ) {
-        let dismissAction = DismissAction(action: { [weak self] in
-            Task {
-                await self?.parseAndValidateAutomaticCaptureKey(key)
+        Task {
+            guard await services.authenticatorItemRepository.isPasswordManagerSyncActive() else {
+                captureCoordinator.navigate(
```

Comment:
🤔 Given that this is in a `Task`, shouldn't the `captureCoordinator` be weak and checked as we do with `self`? Or are we sure we're always going to have such coordinator available here?

Reply:
My initial thought was we should move to capturing it weakly. However, when I try that it actually allows `captureCoordinator` to get dealloc'd, leaving us with an unresponsive UI. It even fails all of the unit tests (which it seems like it wouldn't, given that the tests are passing a strongly referenced `captureCoordinator` into the `subject`).

Given the way this is structured, I think it's probably best to leave it strongly captured. There's no good way to handle it being dealloc'd - since the caller of this method is also expecting it is the delegate's responsibility to handle dismissing.

Reply:
Oh alright, makes sense thanks for the clarification!